File: Data Structures & Algorithms/remove-element/submission-0.cpp
class Solution {
public:
    void removeEntry(vector<int>& nums, int idx, int length) {
        for(int index = idx +1; index<length; index++) {
            nums[index - 1] = nums[index];
        }
        nums.pop_back();
    }
    int removeElement(vector<int>& nums, int val) {
        int idx = 0;
        while(idx < nums.size()) {
            while(nums[idx] == val && idx < nums.size()) {
                removeEntry(nums,idx,nums.size());
            }
            idx++;
        }
        return nums.size();
    }
};

File: Data Structures & Algorithms/find-pivot-index/submission-0.rs
impl Solution {
    pub fn pivot_index(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut sums = vec![0; n];
        let mut prefix = 0;
        let mut postfix = 0;

        (0..n).for_each(|idx| {
            let rev_idx = n - idx - 1;
            prefix += nums[idx];
            postfix += nums[rev_idx];
            sums[idx] += prefix;
            sums[rev_idx] -= postfix;
        });
        for (idx, val) in sums.into_iter().enumerate() {
            if val == 0 {
                return idx as i32;
            }
        }
        -1
    }
}
